### PR TITLE
feat: Add simple nv23 migration

### PR DESCRIPTION
**Pull request by rjan90**
Add simple nv23 migration

### CODE REVIEW

**Comment by rjan90** [comment]
I ran this simple migration on a local devnet. And it worked fine:

**Pre-Migration:**
```
manifest cid: bafy2bzacecn7uxgehrqbcs462ktl2h23u23cmduy2etqj6xrd6tkkja56fna4
2024-05-02T09:59:16.039+0200    INFO    fil-consensus   filcns/upgrades.go:2741 Started 5 workers
2024-05-02T09:59:16.040+0200    INFO    fil-consensus   filcns/upgrades.go:2741 Creating migration jobs
2024-05-02T09:59:16.040+0200    INFO    fil-consensus   filcns/upgrades.go:2741 Result writer started
2024-05-02T09:59:16.040+0200    INFO    fil-consensus   filcns/upgrades.go:2741 Done creating 17 migration jobs after 984.625µs
2024-05-02T09:59:16.040+0200    INFO    fil-consensus   filcns/upgrades.go:2741 Worker 3 done
2024-05-02T09:59:16.040+0200    INFO    fil-consensus   filcns/upgrades.go:2741 Worker 0 done
2024-05-02T09:59:16.040+0200    INFO    fil-consensus   filcns/upgrades.go:2741 Worker 4 done
2024-05-02T09:59:16.040+0200    INFO    fil-consensus   filcns/upgrades.go:2741 Worker 2 done
2024-05-02T09:59:16.040+0200    INFO    fil-consensus   filcns/upgrades.go:2741 Worker 1 done
2024-05-02T09:59:16.041+0200    INFO    fil-consensus   filcns/upgrades.go:2741 Result writer wrote 17 results to state tree after 1.200375ms
2024-05-02T09:59:16.041+0200    INFO    fil-consensus   filcns/upgrades.go:2741 All workers done after 1.194708ms
2024-05-02T09:59:16.041+0200    INFO    fil-consensus   filcns/upgrades.go:2741 All 17 done after 1.384ms (12283/s)
2024-05-02T09:59:16.041+0200    WARN    statemgr        stmgr/forks.go:263      COMPLETED pre-migration {"duration": 0.004097291}
```

**Migration:**
```
2024-05-02T10:07:22.090+0200    WARN    statemgr        stmgr/forks.go:202      STARTING migration      {"height": "200", "from": "bafy2bzaceczcsmczvrmkdpic43tjts65rhzlkhfx75klvnh3o5mjdscionqvq"}
manifest cid: bafy2bzacecn7uxgehrqbcs462ktl2h23u23cmduy2etqj6xrd6tkkja56fna4
2024-05-02T10:07:22.091+0200    INFO    fil-consensus   filcns/upgrades.go:2741 Started 7 workers
2024-05-02T10:07:22.091+0200    INFO    fil-consensus   filcns/upgrades.go:2741 Creating migration jobs
2024-05-02T10:07:22.091+0200    INFO    fil-consensus   filcns/upgrades.go:2741 Result writer started
2024-05-02T10:07:22.091+0200    INFO    fil-consensus   filcns/upgrades.go:2741 Done creating 17 migration jobs after 59.084µs
2024-05-02T10:07:22.091+0200    INFO    fil-consensus   filcns/upgrades.go:2741 Worker 1 done
2024-05-02T10:07:22.091+0200    INFO    fil-consensus   filcns/upgrades.go:2741 Worker 0 done
2024-05-02T10:07:22.091+0200    INFO    fil-consensus   filcns/upgrades.go:2741 Worker 3 done
2024-05-02T10:07:22.091+0200    INFO    fil-consensus   filcns/upgrades.go:2741 Worker 4 done
2024-05-02T10:07:22.091+0200    INFO    fil-consensus   filcns/upgrades.go:2741 Worker 2 done
2024-05-02T10:07:22.091+0200    INFO    fil-consensus   filcns/upgrades.go:2741 Worker 6 done
2024-05-02T10:07:22.091+0200    INFO    fil-consensus   filcns/upgrades.go:2741 Worker 5 done
2024-05-02T10:07:22.091+0200    INFO    fil-consensus   filcns/upgrades.go:2741 All workers done after 106.75µs
2024-05-02T10:07:22.091+0200    INFO    fil-consensus   filcns/upgrades.go:2741 Result writer wrote 17 results to state tree after 115.542µs
2024-05-02T10:07:22.091+0200    INFO    fil-consensus   filcns/upgrades.go:2741 All 17 done after 121.084µs (140398/s)
2024-05-02T10:07:22.091+0200    WARN    statemgr        stmgr/forks.go:211      COMPLETED migration     {"height": "200", "from": "bafy2bzaceczcsmczvrmkdpic43tjts65rhzlkhfx75klvnh3o5mjdscionqvq", "to": "bafy2bzaceczcsmczvrmkdpic43tjts65rhzlkhfx75klvnh3o5mjdscionqvq", "duration": 0.000654625}
```

Network version:
```
lotus state network-version   
Network Version: 23
```

**Review by arajasek — COMMENTED** [review]
The diff here looks a little funny -- can we edit this to just be commit `bf2c73a9f2a189e9661cf34af2cdd790a890aa2f` applied on top of current `master`?

1. `git reset --hard master`
2. `git cherry-pick bf2c73a9f2a189e9661cf34af2cdd790a890aa2f`
3. `git push --force` 

should do it without any conflicts.

**Comment by rjan90** [comment]
Thank you so much for the pointers! It now looks a lot cleaner.

File: builtin/v14/migration/top.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+	newSystemCodeCID, ok := newManifest.Get(manifest.SystemKey)
+	if !ok {
+		return cid.Undef, xerrors.Errorf("code cid for system actor not found in new manifest")
+	}
+
+	migrations[systemActor.Code] = systemActorMigrator{OutCodeCID: newSystemCodeCID, ManifestData: newManifest.Data}
+
+	if len(migrations)+len(deferredCodeIDs) != len(oldManifestData.Entries) {
+		return cid.Undef, xerrors.Errorf("incomplete migration specification with %d code CIDs, need %d", len(migrations), len(oldManifestData.Entries))
```
Comment (arajasek) [inline builtin/v14/migration/top.go:90]:
```suggestion
		return cid.Undef, xerrors.Errorf("incomplete migration specification with %d code CIDs, need %d", len(migrations)+len(deferredCodeIDs), len(oldManifestData.Entries))
```

**Review by arajasek — APPROVED** [review]
LGTM with one nit (just the error message).

**Pull request merged**